private/protocol/jsonrpc/unmarshal_error.go

```diff
@@ -50,18 +70,29 @@ func (u *UnmarshalTypedError) UnmarshalError(
 	code := codeParts[len(codeParts)-1]
 	msg := jsonErr.Message
 
+	queryCodeParts := queryCodeParts(resp, u)
+
 	if fn, ok := u.exceptions[code]; ok {
 		// If exception code is know, use associated constructor to get a value
 		// for the exception that the JSON body can be unmarshaled into.
-		v := fn(respMeta)
+		var v error
+		queryErrFn, queryExceptionsFound := u.queryExceptions[code]
+		if queryCodeParts != nil && len(queryCodeParts) == 2 && queryExceptionsFound {
```

Comment:
can we document this `2` value somewhere? maybe save it to a variable and add a comment, or as a function comment on the `queryCodeParts` function. and if there already exists public documentation about this header in the API then it would be great to link to that.

also, no need to do a `queryCodeParts != nil` check here (same on line 92) since its safe to do `len` on a nil slice. for example, 
```
func main() {
	var queryCodeParts []string = nil
	fmt.Printf("%d", len(queryCodeParts))
}
```
will print 0.
additionally, `strings.Split` will never return a nil slice
https://pkg.go.dev/strings#Split